plugins/woocommerce/includes/rest-api/Controllers/Version4/Products/README.md

```diff
@@ -0,0 +1,45 @@
+# WooCommerce REST API v4 - Products Endpoint
+
+This directory contains the implementation of the WooCommerce REST API v4 Products endpoint (`/wp-json/wc/v4/products`).
+
+## Overview
+
+The v4 Products endpoint starts as a **copy-paste of the v3 implementation** and will be incrementally enhanced with new features. This approach allows for rapid development while maintaining backward compatibility (for now) and enables easy tracking of changes through dedicated PRs.
+
+**Current Status**: Experimental feature requiring the `rest-api-v4` feature flag.
+
+## Architecture
+
+- **Namespace**: `wc/v4`
+- **Future Goal**: Eventually migrate to extend `WP_REST_Controller` directly instead of WooCommerce base classes
```

Comment:
I think we should list implementing the new version under the WooCommerce namespace as a goal, too.

Reply:
Addressed with 1669ac74f6334b5da272937c55cd4cbd3ed62947